### PR TITLE
Update gh actions

### PR DESCRIPTION
**Pull request by utam0k**
*Description of changes:*

A bit improved github actions.

### CODE REVIEW

**Comment by utam0k** [comment]
@kzys Can I ask to rerun the CI? Because I can't think of any reason why CI is failed. I could pass `EXTRAGOARGS='-exec sudo' make test` on my local 🤔

**Comment by Kern--** [comment]
Rerunning CI.

I don't think we've had flakiness out of this test before, but for future investigations if this comes back, there was an error on Ubuntu 18.04 and go 1.17 when running root tests:

```
--- FAIL: TestTaskManager_CreateExecDeleteTask (0.21s)
[77](https://github.com/firecracker-microvm/firecracker-containerd/runs/7961227545?check_suite_focus=true#step:7:78)
    task_test.go:459: 
[78](https://github.com/firecracker-microvm/firecracker-containerd/runs/7961227545?check_suite_focus=true#step:7:79)
        	Error Trace:	task_test.go:459
[79](https://github.com/firecracker-microvm/firecracker-containerd/runs/7961227545?check_suite_focus=true#step:7:80)
        	Error:      	"[&WaitRequest{ID:fakeTask,ExecID:,XXX_unrecognized:[],}]" should have 2 item(s), but has 1
[80](https://github.com/firecracker-microvm/firecracker-containerd/runs/7961227545?check_suite_focus=true#step:7:81)
        	Test:       	TestTaskManager_CreateExecDeleteTask
[81](https://github.com/firecracker-microvm/firecracker-containerd/runs/7961227545?check_suite_focus=true#step:7:82)
        	Messages:   	Wait called unexpected number of times for "fakeTask"
```

https://github.com/firecracker-microvm/firecracker-containerd/runs/7961227545?check_suite_focus=true#step:7:77

**Comment by utam0k** [comment]
@Kern-- hmm... But now this PR passes the CI 🤔 Should I create an issue about the flakiness?

**Comment by kzys** [comment]
seems we already have #566 for the flakiness.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
 
     strategy:
       matrix:
-        os: ['ubuntu-18.04', 'ubuntu-20.04']
+        os: ['ubuntu-18.04', 'ubuntu-20.04', 'ubuntu-22.04']
```
Comment (austinvazquez) [inline .github/workflows/build.yaml:16]: +1, I think it would also be okay to remove `ubuntu-18.04` since it is now deprecated.

Comment (utam0k) [reply .github/workflows/build.yaml]: 👍

**Comment by utam0k** [comment]
@Kern-- @kzys @austinvazquez Thanks for your review and comments. PTAL

**Comment by austinvazquez** [comment]
> @Kern-- @kzys @austinvazquez Thanks for your review and comments. PTAL

LGTM, the DCO check is failing because each commit needs to be signed. 😅

**Comment by utam0k** [comment]
> LGTM, the DCO check is failing because each commit needs to be signed. 😅

@austinvazquez ops, I forgot it. I signed.

**Review by ginglis13 — APPROVED** [review]

**Pull request merged**